src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt

```diff
@@ -402,21 +427,83 @@ class AccountControllerTest @Autowired constructor(
         @Test
         fun `should fail to create account with existing email`() {
             println("testAccount: ${objectMapper.writeValueAsString(testAccount)}")
-            mockMvc.post("/accounts/new") {
-                contentType = MediaType.APPLICATION_JSON
-                content = testAccount.toJson()
+
+            val accountPart = MockPart("account", testAccount.toJson().toByteArray())
+            accountPart.headers.contentType = MediaType.APPLICATION_JSON
+
+            mockMvc.multipart("/accounts/new") {
+                part(accountPart)
```

Comment:
This code is everywhere, what if we extend mockMvc with a new method that takes a string?

Reply:
I think the new methodology using a builder pattern is quite clean

Reply:
I like it too! Good job